org.nickvision.tubeconverter.gnome/src/application.cpp

```diff
@@ -52,15 +53,17 @@ namespace Nickvision::TubeConverter::GNOME
         {
             m_mainWindow = std::make_shared<Views::MainWindow>(m_controller, app);
         }
+    }
+
+    void Application::onActivate(GtkApplication* app)
+    {
         m_mainWindow->show();
     }
 
     void Application::onOpen(GtkApplication* app, void* files, int n, const char* hint)
     {
-        if(n > 0)
-        {
-            GFile** files{ reinterpret_cast<GFile**>(files) };
-            m_mainWindow->addDownload(g_file_get_uri(files[0]));
-        }
+        GFile** arr{ reinterpret_cast<GFile**>(files) };
+        m_mainWindow->show();
+        m_mainWindow->addDownload(g_file_get_uri(arr[0]));
```

Comment:
Although, unlikely that someone would use it this way. It is now possible segfault by calling `Open` with an empty files list because `n` isn't checked to be non-zero.

Reply:
Oops, just a few seconds too late.

Reply:
According to: https://docs.gtk.org/gio/method.Application.open.html

> `n_files must be greater than zero.` 

So the signal can not receive 0, no?

Reply:
> So the signal can not receive 0, no?

If this is not the case I can add the check before releasing...

Reply:
Oh, you are right. If I send an empty array via DBus, it crashes due to some validation error, not because of a segfault.

Reply:
Perfect! Thanks for checking